src/utils/keplr.ts

```diff
@@ -0,0 +1,35 @@
+export const signWithKeplr = async (
+  sommelierAddress: string,
+  ethAddress: string,
+  sommAddress: string
+) => {
+  if (!window.getOfflineSigner || !window.keplr) {
+    throw new Error("Please install Keplr extension")
+  }
+
+  const chainId = "sommelier-3" 
+  await window.keplr.enable(chainId)
+  const signer = window.getOfflineSigner(chainId)
+  const accounts = await signer.getAccounts()
+
+  if (accounts[0].address !== sommelierAddress) {
+    throw new Error(
+      "Keplr account does not match the provided Sommelier address"
+    )
+  }
+
+
+  const message = `Sign this message to link your Ethereum address: ${ethAddress} with your Sommelier address: ${sommAddress}`
+  const signBytes = new TextEncoder().encode(message)
+
+  const { signature } = await signer.signAmino(accounts[0].address, {
```

Comment:
We need to use sign arbitrary here

https://docs.keplr.app/api/#request-signature-for-arbitrary-message

Reply:
instead of sign Amino